src/overlays/actors/ovl_En_Insect/z_en_insect.h

```diff
@@ -17,16 +27,25 @@ typedef struct EnInsect {
     /* 0x01F0 */ Vec3s jointTable[24];
     /* 0x0280 */ Vec3s morphTable[24];
     /* 0x0310 */ EnInsectActionFunc actionFunc;
-    /* 0x0314 */ u16 unk_314;
+    /* 0x0314 */ u16 flags;
     /* 0x0316 */ s16 unk_316;
     /* 0x0318 */ s16 unk_318;
-    /* 0x031A */ s16 unk_31A;
-    /* 0x031C */ s16 unk_31C;
-    /* 0x031E */ s16 unk_31E;
+    /* 0x031A */ s16 actionTimer;
+    /* 0x031C */ s16 lifeTimer;
+    /* 0x031E */ s16 crawlSoundTimer;
     /* 0x0320 */ ObjMakekinsuta* soilActor;
     /* 0x0324 */ f32 unk_324;
     /* 0x0328 */ s16 unk_328;
     /* 0x032A */ u8 unk_32A;
 } EnInsect; // size = 0x032C
 
+typedef enum {
+    /* 0 */ INSECT_PERMANENT,
+    /* 1 */ INSECT_SPAWNED,
+    /* 2 */ INSECT_FIRST_DROPPED,
+    /* 3 */ INSECT_EXTRA_DROPPED // An additional bug spawned by dropping one from a bottle.
+} EnInsectType;
+
+#define IS_DROPPED(type) ((type) == INSECT_FIRST_DROPPED || (type) == INSECT_EXTRA_DROPPED)
```

Comment:
I think this should go in the .c maybe, or if it stays in the .h it should have a less generic name